pkg/metrics/metrics.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2018 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// ServerMetrics contains Prometheus metrics for the Ark server.
+type ServerMetrics struct {
+	metrics map[string]prometheus.Collector
+}
+
+const (
+	metricNamespace      = "ark_server"
+	backupSizeBytesGauge = "backup_size_bytes"
+	backupAttemptsCount  = "backup_attempts_total"
+	backupSuccessCount   = "backup_success_total"
```

Comment:
If we keep total, this name is fine as-is. Otherwise I'd maybe call it `backup_successes`

Also note that we may move away from success/failure for backups. For restores, the valid phases are new, in progress, failed validation, and completed. In each restore's status, we record the number of warnings and errors that occurred, but as long as we're able to restore something, the status is always completed. We are considering doing something similar for backups. (#286 is related)

Reply:
When we make that change, we can call this metric `backup_completed_total`. Or even have a metric per phase of the backup. WDYT?

Reply:
Sure, and we can discuss it if/when we make that change.